src/hooks/useHandleTouches.ts

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { Direction } from '../types';
+
+type Coords = {
+  clientX: number;
+  clientY: number;
+};
+
+const MIN_DISTANCE = 50;
+
+const getSwipeDirection = (startCoords: Coords, endCoords: Coords): Direction | undefined => {
+  const absX = Math.abs(startCoords.clientX - endCoords.clientX);
+  const absY = Math.abs(startCoords.clientY - endCoords.clientY);
+
+  if (absX > absY && absX > MIN_DISTANCE) {
+    return startCoords.clientX > endCoords.clientX ? 'LEFT' : 'RIGHT';
+  } else if (absY > MIN_DISTANCE) {
+    return startCoords.clientY > endCoords.clientY ? 'UP' : 'DOWN';
+  }
+
+  return undefined;
+};
+
+export const useHandleTouches = (onSwipe: (direction: Direction) => void) => {
+  const [startCoords, setStartCoords] = useState<Coords | null>(null);
```

Comment:
`startCoords` doesn't participate in rendering is not a dependency of a hook. It means that it is better to implement it as `Ref`